### PR TITLE
Bugfix/thread synchronization

### PR DESCRIPTION
**Pull request by ChristianMorup**
Refatored file monitoring so it takes synchronization into account. From now on a new listener and projection controller is created whenever a new folder is detected. This prevents the need for synchronization in between different queues.

### CODE REVIEW

File: RTDS/RTDS.UnitTest/Monitoring/Test_BaseFolderController.cs [hunk 1]
```diff
@@ -25,56 +25,40 @@ public void SetUp()
...
-
-        [Test]
-        public void StartMonitoring_NoFolderIsAdded_FileMonitorIsNotCreated()
-        {
-            //Arrange: 
-            var path = _paths.BaseSourcePath;
-
-            //Act:
-            _uut.StartMonitoring();
+                     new SearchDirectoryArgs(filePath, fileName));
```
Comment (Cpilgaard) [inline RTDS/RTDS.UnitTest/Monitoring/Test_BaseFolderController.cs]: Is there a weird tab here? Or is it a flaw?

Comment (ChristianMorup) [reply RTDS/RTDS.UnitTest/Monitoring/Test_BaseFolderController.cs]: It neither - Its how that it suggest formatting when the line is to broad.

File: RTDS/RTDS/DTO/ProjectionInfo.cs [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
+﻿namespace RTDS.DTO
 {
     internal class ProjectionInfo
     {
-        public ProjectionInfo(ProjectionFolderStructure structure)
+        public ProjectionInfo(int id, string permanentStoragePath, string tempStoragePath, string fileName)
         {
-            Structure = structure;
-            FilesToBeTransferred = new BlockingCollection<string>();
-            FilesToBeConverted = new BlockingCollection<string>();
```
[the remark this answers is not in the document]
Comment (ChristianMorup) [reply RTDS/RTDS/DTO/ProjectionInfo.cs]: We are. But they are contained in the ProjectionController instead.

File: RTDS/RTDS.UnitTest/Monitoring/Test_BaseFolderController.cs (hunk 1 above)
Comment (Cpilgaard) [reply RTDS/RTDS.UnitTest/Monitoring/Test_BaseFolderController.cs]: I see it clearly now...

**Pull request merged**